src/acl/AtStep.cc

```diff
@@ -8,25 +8,29 @@
 
 #include "squid.h"
 
-#if USE_OPENSSL
-
 #include "acl/AtStep.h"
 #include "acl/AtStepData.h"
 #include "acl/FilledChecklist.h"
 #include "client_side.h"
 #include "http/Stream.h"
+#if USE_OPENSSL
 #include "ssl/ServerBump.h"
+#endif
 
 int
-ACLAtStepStrategy::match (ACLData<Ssl::BumpStep> * &data, ACLFilledChecklist *checklist)
+ACLAtStepStrategy::match (ACLData<XactionSteps> * &data, ACLFilledChecklist *checklist)
 {
+    Must(checklist->request);
+    Must(checklist->request->masterXaction);
+    if (checklist->request->masterXaction->generatingConnect && data->match(xaStepGeneratingConnect))
```

Comment:
Given the risk we take with a missing `checklist->request->masterXaction`, I think this should become:

```suggestion
    if (data->match(xaStepGeneratingConnect)) {
        if (!checklist->request)
            return 0; // we have warned about the missing request earlier
        ... code warning about treating missing masterXaction as a mismatch ...
        return checklist->request->masterXaction->generatingConnect ? 1 : 0;
    }
```

Reply:
commit 8f212b54d6